### PR TITLE
doc: Explain a little how the virtual keyboard interacts with text input and visibility

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

File: examples/virtual_keyboard/README.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+1. The virtual keyboard itself is implemented in "virtual_keyboard.slint" as a re-usable component. The application is responsible for placing it in the scene, typically as the last item in the root component.
+2. When a `TextInput` element receives the focus, either by the user clicking or programmatically via a call to `focus()`, it sets the global `TextInputInterface.text-input-focused` property to true. Similarly,
+   when the focus is lost, it is set to false again. Use this property to control visibility of the virtual keyboard.
+3. When the user clicks on a key in the virtual keyboard, the application needs to simulate a key event as if the user pressed it on a real keyboard. The virtual keyboard invokes `VirtualKeyboardHandler`'s `key_pressed` callback. Set it to dispatch a key event to the `slint::Window`. Slint takes care of routing it to the currently focused `TextInput`. In Rust, call `slint::Window::dispatch_event(slint::platform::WindowEvent::KeyPressed{...})` to dispatch the event, in C++ call `slint::Window::dispatch_key_press_event`.(...)`. Subsequently, the you should dispatch a key release event using the same functions.
```
Comment (ogoffart) [inline examples/virtual_keyboard/README.md:8]:
```suggestion
3. When the user clicks on a key in the virtual keyboard, the application needs to simulate a key event as if the user pressed it on a real keyboard. The virtual keyboard invokes `VirtualKeyboardHandler`'s `key_pressed` callback. Set it to dispatch a key event to the `slint::Window`. Slint takes care of routing it to the currently focused `TextInput`. In Rust, call `slint::Window::dispatch_event(slint::platform::WindowEvent::KeyPressed{...})` to dispatch the event, in C++ call `slint::Window::dispatch_key_press_event(...)`. Subsequently, the you should dispatch a key release event using the same functions.
```

**Review by ogoffart — APPROVED** [review]
Unrelated to this particular commit, but in general, it seems you have a tendency to write long lines in the .md files. (or html files)
Personally, i find that breaking lines at least on period helps to keep somewhat short lines in source .md files, and also help to keep the diff shorts in the future when sentences are modified
(Also I once was reading this https://news.ycombinator.com/item?id=31808093 )

**Review by FloVanGH — APPROVED** [review]

File: examples/virtual_keyboard/README.md [hunk 2]
```diff
@@ -1,8 +1,11 @@
 # Virtual Keyboard Example
 
-This example application demonstrates how to implement and display a custom virtual keyboard in Slint.
+This example application demonstrates how to implement and display a custom virtual keyboard in Slint. It is composed of three different building blocks:
```
Comment (fieran100) [inline examples/virtual_keyboard/README.md:3]: Maybe : 

"It is composed of .. " --> "It has"

File: examples/virtual_keyboard/README.md (hunk 1 above)
Comment (fieran100) [inline examples/virtual_keyboard/README.md:8]:
```suggestion
1. The virtual keyboard itself: This is implemented in `virtual_keyboard.slint` as a re-usable component. The application is responsible for placing it in the scene, typically as the last item in the root component.
2. Keyboard visibility: When a `TextInput` element receives focus, either by the user clicking on it or programmatically via a call to `focus()`, it sets the global `TextInputInterface.text-input-focused` property to true. Similarly,
   when focus is lost, this property is set to false. Use this property to control visibility of the virtual keyboard.
3. Interaction: When the user clicks on a key in the virtual keyboard, the application needs to simulate a key event as if the user pressed the key on a real keyboard. The virtual keyboard invokes `VirtualKeyboardHandler`'s `key_pressed` callback. You need to set this callback to dispatch a key event to the `slint::Window`. Slint takes care of routing it to the currently focused `TextInput`. In Rust, call `slint::Window::dispatch_event(slint::platform::WindowEvent::KeyPressed{...})` to dispatch the event; in C++ call `slint::Window::dispatch_key_press_event`.(...)`. Subsequently, you should dispatch a key release event using the same family of functions.
```

**Review by fieran100 — COMMENTED** [review]
Hi, I made some suggestions. The idea is to highlight the three buildings blocks and then explain them in more detail. Hope this helps.

**Comment by tronical** [comment]
Thank you everyone. Feedback applied and lines broken to occupy fewer columns :-)

**Pull request merged**